numba/cuda/tests/cudapy/test_atomics.py

```diff
@@ -553,7 +553,7 @@ def test_atomic_and_global(self):
 
         gold = np.zeros(32, dtype=np.int32)
         for i in range(idx.size):
-            gold[idx[i]] &= 1
+            gold[idx[i]] &= rand_const
```

Comment:
Why did this work before?

Reply:
@gmarkall I don't know why it worked before; I racked my brain for several hours yesterday trying to figure this out. It seems to be something particular to the "&" operator. It fails on "|" for example.

Reply:
Does this test even make sense?  (Maybe I'm missing something obvious...)  If `gold` and `ary` are initialized to zeros, won't the result of anding the values with anything result in a zero array at the end?

Reply:
> Does this test even make sense? (Maybe I'm missing something obvious...) If `gold` and `ary` are initialized to zeros, won't the result of anding the values with anything result in a zero array at the end?

@seibert yes that's it; I was staring at this so long I missed the obvious. Updating the tests to non-zero initialized arrays.